aten/src/ATen/cuda/detail/CUDAGuardImpl.cpp

```diff
@@ -0,0 +1,9 @@
+#include <ATen/cuda/detail/CUDAGuardImpl.h>
+
+namespace at {
+namespace cuda {
+namespace detail {
+
+C10_REGISTER_GUARD_IMPL(CUDA, CUDAGuardImpl);
```

Comment:
I feel like we're starting to have too many static registries.

Reply:
The plan on record is to refactor this into a generic interface that other people can use.